27_tictactoe/main.py

```diff
@@ -0,0 +1,46 @@
+import re
+import lib.board as board
+from lib.player import Player
+
+players = [None,None]
+i = 0
+
+while True:
+
+    # Get players' info
+    while i < 2:
+        try:
+            players[i] = Player(name   = raw_input("Player %d - Name:   " %(i)),
+                                marker = raw_input("Player %d - Marker: " %(i)))
+            i += 1
+        except Exception as e:
+            print "Error occured while setting player %d\nError message: %s" %(i, e.message)
+
+
+    # Init game
+    tto = board.Game()
+    tto.start(players[0], players[1])
+
+    # Play game
+    while (tto.winner is None) and (not tto.tie()):
+        current_play = raw_input("%s's play (enter row,col): " % tto.current_player.name)
+        m = re.search(r'^\s*(\w)[,;\s\.]*(\w)\s*$', current_play)
+        try:
+            tto.play(int(m.group(1)),int(m.group(2)))
+            print tto
+        except board.InvalidMove as e:
+            print e.message
+        except Exception as e:
+            print e.message
+
+    # Print results
+    if tto.tie(): print "Game tied!!"
+    else: print "Winner is %s - well done!!\n" %(tto.winner.name)
+    print "------- GAME OVER --------\n"
+
+    # Continue?
+    if raw_input("Would you like to play again (y/n)? ").lower() == 'n': 
+        print "\nGood bye... Hope you enjoyed!!\n"
+        break
+    if raw_input("Same players (y/n)? ").lower() == 'n': 
+        i = 0
```

Comment:
Problem was not in Player class but in `tie` detection. In the test `play` method is expected to get (col, row), and in your game it takes (row, col). That's why the play sequence in the test does not product a tie and the test fails.

This is the sequence that according to the test was to product a tie, but in your game produces victory for the first player:

 28         self.g.play(0,0)
 29         self.g.play(0,1)
 30         self.g.play(0,2)
 31         self.g.play(1,1)
 32         self.g.play(1,0)
 33         self.g.play(1,2)
 34         self.g.play(2,1)
 35         self.g.play(2,2)
 36         self.g.play(2,0)

That's not too bad, just solving in a different way.